definitions/npm/ramda_v0.x.x/flow_v0.82.x-/test_ramda_v0.x.x_filter.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+import R, { filter, type RefineFilter } from "ramda";
+import { it, describe } from "flow-typed-test";
+
+describe("filter", () => {
+  it("perserves the element type in the predicate and result array (number)", () => {
+    const s: Array<number> = filter(x => x > 1, [1, 2]);
+  });
+
+  it("perserves the element type in the predicate and result array (number)", () => {
+    const s: Array<string> = R.filter(x => x === "2", ["2", "3"]);
+  });
+
+  it("filters objects by passing the value to the predicate", () => {
+    const s: { [key: string]: string } = R.filter(x => x === "2", {
+      a: "2",
+      b: "3"
+    });
+  });
+
+  it("refines the element type using the predicate", () => {
+    const notNull = (x): boolean %checks => x != null;
+    const ns: Array<number> = (filter: RefineFilter)(notNull, [1, 2, null]);
+  });
+
+  it("fails when type refinement is incorrect", () => {
+    const isString = (x): boolean %checks => typeof x === "string";
+    // $ExpectError
+    const ns: Array<number> = (filter: RefineFilter)(isString, ["1", 2]);
+  });
+
+  it("fails when attempting to refine from a non $Pred predicate", () => {
+    const isNumber = x => typeof x === "number";
+    // $ExpectError
+    const ns: Array<number> = filter(isNumber, ["1", 2]);
+  });
+
+  it("does not accept predicates missing %checks when using RefineFilter", () => {
+    const isNumber = x => typeof x === "number";
+    // $ExpectError
+    const ns: Array<number> = (filter: RefineFilter)(isNumber, ["1", 2]);
+  });
+
+  describe("read only", () => {
+    it("should return readonly Array<number>", () => {
+      const readOnlyArrNumbers: $ReadOnlyArray<number> = Object.freeze([1, 2, 3]);
+
+      (filter(x => x > 1, readOnlyArrNumbers): $ReadOnlyArray<number>);
+
+      (filter(x => x > 1)(readOnlyArrNumbers): $ReadOnlyArray<number>);
```

Comment:
Given my prior comment about sum types with the inputs on `filter`, I'm not sure how this could pass. As I understand it, this means the result is being cast from a `Array<number>` to a `$ReadOnlyArray<number>` and Flow is just allowing that to happen. Am I missing something?